### PR TITLE
Added IssueLogParts and SupplierParts tables

### PR DESCRIPTION
**Pull request by VitaliiMaksymiv**

### CODE REVIEW

**Review by konerokua — DISMISSED** [review]

**Review by konerokua — DISMISSED** [review]

**Review by andriyshevchenko — DISMISSED** [review]

**Review by andriyshevchenko — COMMENTED** [review]
Vitalik Please merge dev into your branch because it deleted SearchExpressionAsync method from repository

**Comment by VitaliiMaksymiv** [comment]
> Vitalik Please merge dev into your branch because it deleted SearchExpressionAsync method from repository

Done!

**Review by andriyshevchenko — DISMISSED** [review]

**Review by Vladacdc — APPROVED** [review]

File: TransIT.DAL/Repositories/ImplementedRepositories/IssueLogRepository.cs [hunk 1]
```diff
@@ -60,6 +60,7 @@ public IssueLogRepository(TransITDBContext context)
             .Include(d => d.OldState)
             .Include(e => e.Supplier)
             .Include(x => x.Document)
+            .Include(x => x.IssueLogParts)
```
Comment (Vladacdc) [inline TransIT.DAL/Repositories/ImplementedRepositories/IssueLogRepository.cs:61]: Why you are not including `Part` after including `IssueLogParts` ? How engineer will add parts to issuelog if he would not be able to see what he had already added. Look at other repositories changed and include entities.

**Review by Vladacdc — CHANGES_REQUESTED** [review]

File: TransIT.DAL/Models/Entities/IssueLogParts.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace TransIT.DAL.Models.Entities
+{
+    public class IssueLogParts
```
Comment (igolovchuk) [inline TransIT.DAL/Models/Entities/IssueLogParts.cs]: rename Entity to IssueLogPart ,
in the singular, without **s**

File: TransIT.DAL/Models/Entities/SuppliersParts.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace TransIT.DAL.Models.Entities
+{
+    public class SuppliersParts
```
Comment (igolovchuk) [inline TransIT.DAL/Models/Entities/SuppliersParts.cs]: rename Entity to SuppliersPart ,
in the singular, without s

**Review by igolovchuk — CHANGES_REQUESTED** [review]
please read comments in files

File: TransIT.DAL/Models/Entities/IssueLog.cs [hunk 4]
```diff
@@ -31,6 +31,8 @@ public IssueLog()
         public virtual State NewState { get; set; }
         public virtual State OldState { get; set; }
         public virtual Supplier Supplier { get; set; }
+
         public virtual ICollection<Document> Document { get; set; }
+        public virtual ICollection<IssueLogPart> IssueLogPart { get; set; }
```
Comment (igolovchuk) [inline TransIT.DAL/Models/Entities/IssueLog.cs]: this property name for collection should not be IssueLogPart, should be IssueLogPart**s**

File: TransIT.DAL/Models/Entities/Part.cs [hunk 5]
```diff
@@ -19,5 +20,8 @@ public class Part : IAuditableEntity, IBaseEntity
         public Manufacturer Manufacturer{ get; set; }
         public virtual User Create { get; set; }
         public virtual User Mod { get; set; }
+
+        public ICollection<IssueLogPart> IssueLogPart { get; set; }
+        public ICollection<SupplierPart> SupplierPart { get; set; }
```
Comment (igolovchuk) [inline TransIT.DAL/Models/Entities/Part.cs]: this property names for collections should not be in singular, should be IssueLogPart**s** and SupplierPart**s**

File: TransIT.DAL/Models/Entities/Supplier.cs [hunk 6]
```diff
@@ -26,6 +26,8 @@ public Supplier()
         public virtual Currency Currency { get; set; }
         public virtual User Create { get; set; }
         public virtual User Mod { get; set; }
+
         public virtual ICollection<IssueLog> IssueLog { get; set; }
+        public virtual ICollection<SupplierPart> SupplierPart { get; set; }
```
Comment (igolovchuk) [inline TransIT.DAL/Models/Entities/Supplier.cs]: SupplierParts

**Review by igolovchuk — CHANGES_REQUESTED** [review]
please read comments in files and be more attentive, I did not ask to rename navigation properties, only dbentities into singular form

File: TransIT.DAL/Models/Entities/SuppliersParts.cs (hunk 3 above)
Comment (BohdanBuhrii) [reply TransIT.DAL/Models/Entities/SuppliersParts.cs]: Better rename to "SupplierPart"

**Review by Vladacdc — APPROVED** [review]

**Review by andriyshevchenko — APPROVED** [review]

**Review by igolovchuk — APPROVED** [review]

**Pull request merged**